### PR TITLE
[extension/dbstorage] support for Postgres

### LINKED ISSUE #35692: [extension/storage/dbstorage] pgx support not wokring because db queries are sqlite specific
### Component(s)

extension/storage/dbstorage

## Description
The `dbstorage` extension says that both `sqlite3` and `pgx` are supported.
> `driver`: the name of the database driver to use. By default, the storage client supports "sqlite3" and "pgx".

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.110.0/extension/storage/dbstorage/README.md?plain=1#L21

However when using the `pgx` driver, the create table statement uses syntax specific to `sqlite3` which causes a failure
> error   graph/graph.go:425      Failed to start component       {"error": "error connecting to storage: ERROR: type \"blob\" does not exist (SQLSTATE 42704)", "type": "Receiver", "id": "sqlquery"}

The queries in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/storage/dbstorage/client.go#L20-L23 look specific to `sqlite3`.

## Steps to Reproduce

See collector configuration below

## Expected Result

I would expect there not to be an error and that the tracking table be created

## Actual Result

> error   graph/graph.go:425      Failed to start component       {"error": "error connecting to storage: ERROR: type \"blob\" does not exist (SQLSTATE 42704)", "type": "Receiver", "id": "sqlquery"}

### Collector version

v0.110.0
Custom collector distro running on alpine compiled with go 1.22

### OpenTelemetry Collector configuration

```yaml
receivers:
  sqlquery:
    driver: postgres
    datasource: "host=localhost port=5432 user=postgres password=s3cr3t sslmode=disable"
    queries:
      - sql: "select * from my_logs where log_id > $$1"
        tracking_start_value: "1" # Not sure if this is correct yet but it should not matter for this issue
        tracking_column: log_id
        logs:
          - body_column: log_body

exporters:
  debug:
    verbosity: detailed

extensions:
  db_storage:
    driver: pgx
    datasource: "host=localhost port=5432 user=postgres password=s3cr3t sslmode=disable"

service:
  extensions: [db_storage]
  pipelines:
    logs:
      receivers: [sqlquery]
      exporters: [debug]
```

### Log output

```shell
error   graph/graph.go:425      Failed to start component       {"error": "error connecting to storage: ERROR: type \"blob\" does not exist (SQLSTATE 42704)", "type": "Receiver", "id": "sqlquery"}
```

### Additional context

Once this is fixed I think there may also be an issue with the tracking table name as I think it will include invalid characters (at least when using sqlqueryreceiver). This is inferring the table name from the code which I might not being doing right. I can't test with sqlite3 due to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35280

### PR DESCRIPTION
**Pull request by atoulme**
#### Description
Adds support for Postgres, using a test running a Postgresql docker image.

#### Link to tracking issue
Fixes #35692

#### Testing
Run a postgresql image and test with it.

### CODE REVIEW

**Review by djaglowski — APPROVED** [review]

File: extension/storage/dbstorage/extension_test.go [hunk 1]
```diff
@@ -112,6 +123,48 @@ func newTestExtension(t *testing.T) storage.Extension {
...
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		require.NoError(t, ctr.Terminate(context.Background()))
+	})
+	f := NewFactory()
+	cfg := f.CreateDefaultConfig().(*Config)
+	cfg.DriverName = "pgx"
+	cfg.DataSource = fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s sslmode=disable", "127.0.0.1", port.Port(), "root", "passwd", "db")
+
+	extension, err := f.CreateExtension(context.Background(), extensiontest.NewNopSettings(), cfg)
```
Comment (dmitryax) [inline extension/storage/dbstorage/extension_test.go:159]: It'd be nice to do some set/get on the storage after initialization

File: extension/storage/dbstorage/client.go [hunk 2]
```diff
@@ -17,10 +17,11 @@ import (
 )
 
 const (
-	createTable     = "create table if not exists %s (key text primary key, value blob)"
-	getQueryText    = "select value from %s where key=?"
-	setQueryText    = "insert into %s(key, value) values(?,?) on conflict(key) do update set value=?"
-	deleteQueryText = "delete from %s where key=?"
+	createTableSqlite = "create table if not exists %s (key text primary key, value blob)"
+	createTable       = "create table if not exists %s (key text primary key, value text)"
```
Comment (dmitryax) [inline extension/storage/dbstorage/client.go:21]: Why text? The getters/setters on the storage extension interface work on `[]byte` type... Doesn't Postgres have a blob type? Maybe it's called differently?

Comment (atoulme) [reply extension/storage/dbstorage/client.go:21]: postgres has no blob type, a bytea type that is not available by default, and text which works well across dbs. I picked text as it is closest.

File: extension/storage/dbstorage/extension_test.go (hunk 1 above)
Comment (atoulme) [reply extension/storage/dbstorage/extension_test.go:159]: yes, we do that in the test, this code sets up the storage client.

**Review by mwear — APPROVED** [review]

**Comment by andrzej-stencel** [comment]
@atoulme please resolve conflicts and we'll merge.

**Comment by atoulme** [comment]
@andrzej-stencel rebased thanks.

**Pull request merged**